root/genre/GenreEditForm.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {withCatalystContext} from '../context';
+import FormRowTextLong from '../components/FormRowTextLong';
+import FormSubmit from '../components/FormSubmit';
+
+import type {GenreFormT} from './types';
+
+type Props = {|
+  +$c: CatalystContextT,
+  +form: GenreFormT,
+|};
+
+const GenreEditForm = ({$c, form}: Props) => (
+  <form action={$c.req.uri} method="post">
+    <div className="half-width">
+      <fieldset>
+        <legend>{l('Genre details')}</legend>
+        <FormRowTextLong
+          field={form.field.name}
+          label={addColonText(l('Name'))}
+          required
+        />
+        <FormRowTextLong
+          field={form.field.comment}
+          label={addColonText(l('Disambiguation'))}
+        />
+      </fieldset>
+    </div>
+    <div className="row no-label">
+      {$c.action.name === 'create' ? (
+        <FormSubmit label={l('Create genre')} />
+      ) : (
+        <FormSubmit label={l('Update genre')} />
+      )}
```

Comment:
These labels are temporary and could be replaced with already replaced label `'Submit'`.
When genres have edit types, this will likely be replaced with label `'Enter edit'`.

Reply:
Done, and added a TODO linking to https://tickets.metabrainz.org/browse/MBS-10165

Reply:
Thanks!